crates/ruff/src/args.rs

```diff
@@ -491,97 +506,168 @@ impl From<&LogLevelArgs> for LogLevel {
     }
 }
 
+#[derive(Default)]
+pub struct ConfigArgs {
+    /// Path to a pyproject.toml or ruff.toml config file (etc.), if provided
+    config_file: Option<PathBuf>,
+    /// Overrides provided via the `--config FOO=BAR` option on the CLI
+    config_overrides: Configuration,
+    /// Overrides provided via specific (legacy-ish) flags such as --line-length etc
+    cli_overrides: CliOverrides,
+}
+
+impl ConfigArgs {
+    pub fn config_file(&self) -> Option<&PathBuf> {
+        self.config_file.as_ref()
+    }
+
+    fn from_cli_options(
+        config_options: Option<Vec<ConfigOption>>,
+        cli_overrides: CliOverrides,
+        isolated: bool,
+    ) -> Result<Self, anyhow::Error> {
+        let mut new = Self {
+            cli_overrides,
+            ..Self::default()
+        };
+
+        let Some(options) = config_options else {
+            return Ok(new);
+        };
+
+        for option in options {
+            match option {
+                ConfigOption::ConfigOverride(overridden_option) => {
+                    new.config_overrides = new.config_overrides.combine(
+                        Configuration::from_options(*overridden_option, &path_dedot::CWD)?,
+                    );
+                }
+                ConfigOption::PathToConfigFile(path) => {
+                    if isolated {
+                        let context = format!(
+                            "Both `--isolated` and `--config={}` were specified on the command line",
+                            path.display()
+                        );
+                        let error = anyhow!(
+                            "Cannot specify `--isolated` and also specify a configuration file"
+                        )
+                        .context(context);
+                        return Err(error);
+                    }
+                    match new.config_file {
+                        None => new.config_file = Some(path),
+                        Some(ref config_file) => {
+                            if config_file == &path {
+                                // I'm not sure why you'd specify the same config file twice,
+                                // but it seems reasonable to let it pass if we do encounter that
+                                continue;
+                            }
```

Comment:
I'm not sure if it's worth special casing here. Seems like unnecessary complexity to handle a very specific, uncommon edge case.

Reply:
Deleted it in 721653de5580ce0c1c192f2625e971d2529f7b03 :)